recipes/llvm-core/all/conanfile.py

```diff
@@ -1,429 +1,496 @@
-                json.dump(components, components_file, indent=4)
-        else:
-            suffixes = ['.dylib', '.so']
-            for name in os.listdir(lib_path):
-                if not any(suffix in name for suffix in suffixes):
-                    os.remove(os.path.join(lib_path, name))
+    @property
+    def _cmake_module_path(self):
+        return Path("lib") / "cmake" / "llvm"
 
-    def package_info(self):
-        self.cpp_info.set_property("cmake_file_name", "LLVM")
+    @property
+    def _build_info_file(self):
+        return Path(self.package_folder) / self._cmake_module_path / "conan_llvm_build_info.json"
+
+    @property
+    def _build_module_file_rel_path(self):
+        return self._cmake_module_path / f"conan-official-{self.name}-variables.cmake"
+
+    def _create_cmake_build_module(self, build_info, module_file):
+        targets_with_jit = ["X86", "PowerPC", "AArch64", "ARM", "Mips", "SystemZ"]
+        content = textwrap.dedent(f"""\
+            set(LLVM_TOOLS_BINARY_DIR "${{CMAKE_CURRENT_LIST_DIR}}/../../../bin")
+            cmake_path(NORMAL_PATH LLVM_TOOLS_BINARY_DIR)
+            set(LLVM_PACKAGE_VERSION "{self.version}")
+            set(LLVM_AVAILABLE_LIBS "{';'.join(build_info['components'].keys())}")
+            set(LLVM_BUILD_TYPE "{self.settings.build_type}")
+            set(LLVM_CMAKE_DIR "${{CMAKE_CURRENT_LIST_DIR}}")
+            set(LLVM_ALL_TARGETS "{self._all_targets}")
+            set(LLVM_TARGETS_TO_BUILD "{self._targets_to_build}")
+            set(LLVM_TARGETS_WITH_JIT "{';'.join(targets_with_jit)}")
+            set(LLVM_NATIVE_ARCH "{build_info['native_arch']}")
+            set_property(GLOBAL PROPERTY LLVM_TARGETS_CONFIGURED On)
+            set_property(GLOBAL PROPERTY LLVM_COMPONENT_LIBS ${{LLVM_AVAILABLE_LIBS}})
+            if (NOT TARGET intrinsics_gen)
+              add_custom_target(intrinsics_gen)
+            endif()
+            if (NOT TARGET omp_gen)
+              add_custom_target(omp_gen)
+            endif()
+            if (NOT TARGET acc_gen)
+              add_custom_target(acc_gen)
+            endif()
+           """)
+        save(self, module_file, content)
+
+    def _write_build_info(self):
+        build_info = self._llvm_build_info()
+        with open(self._build_info_file, "w", encoding="utf-8") as fp:
+            json.dump(build_info, fp, indent=2)
+
+        return build_info
+
+    def _read_build_info(self) -> dict:
+        with open(self._build_info_file, encoding="utf-8") as fp:
+            return json.load(fp)
+
+    def package(self):
+        copy(self, "LICENSE.TXT", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
 
+        build_info = self._write_build_info()
+
+        package_folder = Path(self.package_folder)
+        cmake_folder = package_folder / "lib" / "cmake" / "llvm"
+        rm(self, "LLVMConfig.cmake", cmake_folder)
```

Comment:
I tend to say that we should pass everything to the Conan methods as `Path().as_posix()` instead of `Paths` objects to avoid any unexpected errors.